x-pack/plugins/apm/server/lib/helpers/create_es_client/create_apm_event_client/get_request_base.ts

```diff
@@ -28,7 +28,11 @@ export function processorEventsToIndex(
   events: ProcessorEvent[],
   indices: ApmIndicesConfig
 ) {
-  return uniq(events.map((event) => indices[processorEventIndexMap[event]]));
+  return uniq(
+    events.flatMap((event) =>
+      indices[processorEventIndexMap[event]].split(',').map((str) => str.trim())
+    )
+  );
 }
 
 export function getRequestBase(options: {
```

Comment:
`getRequestBase` is very fuzzy. What about splitting it into `getApmEvents`, `getApmIndices` and `getApmFilters` or something like that?

Reply:
Had a quick look, but the filters depend on `events` (so I think it's better off in a single function that controls how these are merged), and `getApmIndex` is essentially already a separate function (`processorEventsToIndex`).